### DISCUSSION TITLE
Train loosing group after first refuel

### DISCUSSION THREAD

**Discussion by kosinal** [Opinionated-Blueprints/10-Books-Full-of-Rails#102] [Q&A] [answered]
I am most likely creating train in a wrong way, but I have not able to figure out the correct way how to do it.

- When I create train from train blueprint, I do select a group, put some fuel into it.
- After this, train goes into FuelStop1 and to refuelling station for its group.
- After first refuelling, the train lost its group (actually its name get cleared) and I cannot figure out why.

- I put back the name (choosing from other existing trains) back again and everything works fine.

It seems that train are not loosng its name when they got refuelled again, but I do have hard time to find that out.

Thank you much for your help.

Also also I would like to express higly appriciation for your work. Thank you very much for it.

**Comment by Bocian-1** [comment]
Sounds like a bug in the game. I've had trains randomly going into manual for no apparent reason recently and someone else also reported similar behaviour.

**Reply by OldieButGoldie** [reply]
The going manual happened to me, incidentally, before discovering your book, I was using a similar train pool setup to yours: trains will go to a loading station to fill up on cargo, then go to a unloading station to well, unload.

However I missed the wait condition on the unloading interrupt, so the interrupts where evaluated in order by the game: The cargo was not empty, so it couldn't go to the provider station; and the cargo wasn't full either (some inserters started to unload). My conditions were setup like "if cargo empty -> go to provider, wait  until full; if cargo full -> go to sink" The missing wait condition on the second interrupt, and no normal stations in schedule, causes the trains to go to manual.

Perhaps this could be handled more elegantly by the game engine, a warning would be great.

Sorry again for the offtopic, but perhaps it helps you fix an issue.

**Comment by kosinal** [comment, marked as answer]
I would like to apologize. It is not bug neither in game nor in this blueprint. The problem was actually in a clash with TSM mod, which autofuel feature did this.
